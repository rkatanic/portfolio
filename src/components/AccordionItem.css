.accordion-item {
  background: var(--color-input);
  margin: 0 0 1rem;
  padding: 0 2rem;
  display: flex;
  flex-direction: column;
  box-shadow: 0 0 10px hsla(var(--color-hue), 10%, 10%, 0.08);
}

.accordion-item:hover .icon {
  fill: var(--color-primary);
}

.accordion-item-label {
  display: flex;
  align-items: center;
  justify-content: space-between;
  transition: 0.25s ease;
  cursor: pointer;
  font-weight: 500;
}

.accordion-item-label .icon {
  margin-left: 1rem;
  transform: rotate(180deg);
  transition: 0.5s ease;
}

.accordion-item-active .icon {
  transform: rotate(0deg);
}

.accordion-item-label-headline {
  color: var(--color-title);
}

.accordion-item-text {
  margin: 0 2rem 0 0;
  max-height: 0;
  overflow: hidden;
  text-align: justify;
  font-size: 0.925rem;
  transition: max-height 0.5s;
}

.accordion-item-text-open {
  transition: max-height 1s ease-in-out;
  max-height: 100vh;
}
