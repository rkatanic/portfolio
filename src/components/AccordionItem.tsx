import { ReactComponent as CloseIcon } from "../assets/icons/arrow-up.svg";

import "./AccordionItem.css";

interface Props {
  label: string;
  text: string | JSX.Element;
  active: boolean;
  onToggle: () => void;
}

const AccordionItem = ({
  label,
  text,
  active,
  onToggle,
}: Props): JSX.Element => (
  <li
    className={`accordion-item ${active ? "accordion-item-active" : ""}`}
    onClick={onToggle}
  >
    <div className="accordion-item-label">
      <p className="accordion-item-label-headline">
        {label}
        <span className="primary">?</span>
      </p>
      <CloseIcon />
    </div>
    <p
      className={`accordion-item-text ${
        active ? "accordion-item-text-open" : ""
      }`}
    >
      {text}
      <br />
      <br />
    </p>
  </li>
);

export default AccordionItem;
