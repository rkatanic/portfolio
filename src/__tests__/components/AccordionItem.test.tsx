import { fireEvent, getByRole, render } from "@testing-library/react";
import AccordionItem from "../../components/AccordionItem";

describe("AccordionItem", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <AccordionItem
        label="test label"
        text="test text"
        onToggle={jest.fn()}
        active={false}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should render open accordion item", (): void => {
    const { baseElement } = render(
      <AccordionItem
        label="test label"
        text="test text"
        onToggle={jest.fn()}
        active={true}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should handle onToggle when clicked", (): void => {
    const mockOnToggle = jest.fn();
    const { getByText } = render(
      <AccordionItem
        label="test label"
        text="test text"
        onToggle={mockOnToggle}
        active={true}
      />
    );

    fireEvent.click(getByText("test label"));

    expect(mockOnToggle).toHaveBeenCalledTimes(1);
  });
});
