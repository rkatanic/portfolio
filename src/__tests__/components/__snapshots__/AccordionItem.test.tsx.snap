// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItem should render 1`] = `
<body>
  <div>
    <li
      class="accordion-item "
    >
      <div
        class="accordion-item-label"
      >
        <p
          class="accordion-item-label-headline"
        >
          test label
          <span
            class="primary"
          >
            ?
          </span>
        </p>
        <svg>
          arrow-up.svg
        </svg>
      </div>
      <p
        class="accordion-item-text "
      >
        test text
        <br />
        <br />
      </p>
    </li>
  </div>
</body>
`;

exports[`AccordionItem should render open accordion item 1`] = `
<body>
  <div>
    <li
      class="accordion-item accordion-item-active"
    >
      <div
        class="accordion-item-label"
      >
        <p
          class="accordion-item-label-headline"
        >
          test label
          <span
            class="primary"
          >
            ?
          </span>
        </p>
        <svg>
          arrow-up.svg
        </svg>
      </div>
      <p
        class="accordion-item-text accordion-item-text-open"
      >
        test text
        <br />
        <br />
      </p>
    </li>
  </div>
</body>
`;
